### PR TITLE
samples(storage): add sample to print bucket notification

### PR DESCRIPTION
**Pull request by NivedhaSenthil**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [x] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [x] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

File: google-cloud-storage/samples/acceptance/notification_test.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    before :all do
+      @notification = bucket.create_notification topic.name
+    end
+
+    after :all do
+      notification.delete
+    end
+
+    it "Print Notification" do
+      expected_output = "Notification ID: #{notification.id}" + "\n" \
```
Comment (dazuma) [inline google-cloud-storage/samples/acceptance/notification_test.rb:51]: Don't need the string concatenations on these lines. Just do:

```ruby
"Notification ID: #{notification.id}\n" \
```

Alternately (and probably better), use heredoc form:

```ruby
      expected_output = <<~OUTPUT
        Notification ID: #{notification.id}
        Topic Name: #{notification.topic}
        Event Types: #{notification.event_types}
        Kind of Notification: #{notification.kind}
        Custom Attributes: #{notification.custom_attrs}
        Payload Format: #{notification.payload}
        Blob Name Prefix: #{notification.prefix}
        Self Link: #{notification.api_url}
      OUTPUT
```

File: google-cloud-storage/samples/storage_print_pubsub_bucket_notification.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+# [START storage_print_pubsub_bucket_notification]
+def print_pubsub_bucket_notification bucket_name:, notification_id:
+  # The ID of your GCS bucket
+  # bucket_name = "your-unique-bucket-name"
+
+  # The ID of your notification configured for the bucket
+  # notification_id = "your-notification-id"
+
+  require "google/cloud/storage"
```
Comment (dazuma) [inline google-cloud-storage/samples/storage_print_pubsub_bucket_notification.rb:23]: I think we should move the require outside the method (but still inside the region tags). It's much more common for requires to be executed once at the top of a file, outside the class/method definitions, than for them to be executed inside a method like this.

**Review by dazuma — CHANGES_REQUESTED** [review]

File: google-cloud-storage/samples/storage_print_pubsub_bucket_notification.rb (hunk 2 above)
Comment (NivedhaSenthil) [reply google-cloud-storage/samples/storage_print_pubsub_bucket_notification.rb]: True that, have updated it for this sample. We might have to update other samples as well to reflect the same.

**Review by bajajneha27 — APPROVED** [review]
LGTM

**Review by dazuma — APPROVED** [review]

**Pull request merged**